kibicara/platforms/twitter/webapi.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (C) 2020 by Cathy Hu <cathy.hu@fau.de>
+#
+# SPDX-License-Identifier: 0BSD
+
+from fastapi import APIRouter, Depends, HTTPException, Response, status
+from kibicara.config import config
+from kibicara.platforms.twitter.bot import spawner
+from kibicara.platforms.twitter.model import Twitter
+from kibicara.webapi.hoods import get_hood
+from logging import getLogger
+from sqlite3 import IntegrityError
+from ormantic.exceptions import NoMatch
+from peony.oauth_dance import get_oauth_token, get_access_token
+
+
+logger = getLogger(__name__)
+
+
+async def get_twitter(twitter_id: int, hood=Depends(get_hood)):
+    try:
+        return await Twitter.objects.get(id=twitter_id, hood=hood)
+    except NoMatch:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)
+
+
+router = APIRouter()
+twitter_callback_router = APIRouter()
+
+
+@router.get('/')
+async def twitter_read_all(hood=Depends(get_hood)):
+    return await Twitter.objects.filter(hood=hood).all()
+
+
+@router.get('/{twitter_id}')
+async def twitter_read(twitter=Depends(get_twitter)):
+    return twitter
+
```

Comment:
Just curious - what are these functions good for? I don't know much about REST APIs.

Also it seems to me that they don't really differ, as there should not be more than one twitter bot per hood? unless we want to enable admins to add more than one.

Reply:
read all = gives you all twitter bots for a city
read = gives you a twitter bot by id for a city

Reply:
thanks for the information :)